ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/AgentTool.java

```diff
@@ -135,4 +138,15 @@ public String getDefaultVersion() {
             return null;
         }
     }
+
+    private void extractFromChatParameters(Map<String, String> parameters) {
+        if (parameters.containsKey("input")) {
+            try {
+                Map<String, String> chatParameters = gson.fromJson(parameters.get("input"), Map.class);
+                parameters.putAll(chatParameters);
```

Comment:
Why do this ? Looks it's possible to override the original params in `parameters` if they have same key